src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -1253,5 +1256,67 @@ CHIP_ERROR VerifyCertificateSigningRequestFormat(const uint8_t * csr, size_t csr
     return CHIP_NO_ERROR;
 }
 
+bool PemEncoder::NextLine(char * destStr, size_t destSize)
+{
+    StringBuilderBase stringBuilder(mInternalStringBuf, sizeof(mInternalStringBuf));
+    bool hasMore        = false;
+    bool bufferTooSmall = destSize < kMinLineBufferSize;
+
+    switch (mState)
+    {
+    case State::kPrintHeader: {
+        if (!bufferTooSmall)
+        {
+            // The `.48s` is to make sure the header is not wider than out internal string buffer. We clamp the header.
+            stringBuilder.AddFormat("-----BEGIN %.48s-----", mEncodedElement);
+        }
+        mState  = mDerBytes.empty() ? State::kPrintFooter : State::kPrintBody;
+        hasMore = true;
+        break;
+    }
+    case State::kPrintBody: {
+        size_t remaining      = mDerBytes.size() - mProcessedBytes;
+        size_t chunkSizeBytes = std::min(remaining, static_cast<size_t>(kNumBytesPerLine));
+        if (!bufferTooSmall)
+        {
+            size_t encodedLen = static_cast<size_t>(
+                Base64Encode(mDerBytes.data() + mProcessedBytes, static_cast<uint16_t>(chunkSizeBytes), mInternalStringBuf));
+            VerifyOrDie(encodedLen < kMinLineBufferSize);
```

Comment:
```suggestion
            VerifyOrDie(encodedLen < sizeof(mInternalStringBuf));
```
to more directly tie it to the actual assumption the line below makes, right?